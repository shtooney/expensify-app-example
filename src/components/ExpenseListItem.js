import React from 'react';
import {Link} from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';

const ExpenseListItem = ({description, amount, createdAt, id}) => (    
    <div>
        <Link to={`/edit/${id}`} exact={true}>
            <h3>{description}</h3>
        </Link>
        <p>
            {numeral(amount).format('$0,0.00')}
             - 
            {moment(createdAt).format('MMMM Do, YYYY')}
        </p>        
    </div>
);

export default ExpenseListItem;