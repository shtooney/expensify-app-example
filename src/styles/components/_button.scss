.button {    
    background: $blue;
    border: none;
    color: white;
    display: inline-block;
    font-size: $font-size-large;
    padding: $s-size;
    font-weight: 300;
    text-decoration: none;
    line-height: 1;
}

.button--link {    
    background: none;
}

.button--secondary {    
    background: #888888;
}