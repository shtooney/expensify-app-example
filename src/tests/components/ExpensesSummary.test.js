import React from 'react';
import {shallow} from 'enzyme';
import {ExpensesSummary} from '../../components/ExpensesSummary';

test('should render summary for 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={235}/>);
    expect(wrapper).toMatchSnapshot();
});

test('should render total for all expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={23} expensesTotal={235342343}/>);
    expect(wrapper).toMatchSnapshot();
});