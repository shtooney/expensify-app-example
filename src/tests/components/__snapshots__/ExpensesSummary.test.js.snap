// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render summary for 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $2.35
  </h1>
</div>
`;

exports[`should render total for all expenses 1`] = `
<div>
  <h1>
    Viewing 
    23
     
    expenses
     totalling 
    $2,353,423.43
  </h1>
</div>
`;
